### PR TITLE
Improve phing build script and related scripts for improved cross-platform support and error handling

### PR DESCRIPTION
**Pull request by EreMaijala**
TODO:
- [x] Resolve [VUFIND-1478](https://vufind.org/jira/browse/VUFIND-1478) when merging

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -439,18 +466,44 @@
...
+    <if>
+      <available file="${srcdir}/solr" type="dir" />
+      <then>
+        <delete dir="${srcdir}/solr" includeemptydirs="true" failonerror="true" />
+      </then>
+    </if>
+    <if>
+      <available file="${srcdir}/composer.phar" />
+      <then>
+        <delete file="${srcdir}/composer.phar" failonerror="false" />
```
Comment (demiankatz) [inline build.xml]: Since it's inside the available check, we could probably add failonerror="true" here too...

File: build.xml [hunk 2]
```diff
@@ -315,9 +328,8 @@
       <istrue value="${apachectl}" />
       <then>
         <copy file="${srcdir}/local/httpd-vufind.conf" tofile="${apacheconfdir}/vufindtest.conf" />
-        <exec command="${apachectl} restart" outputProperty="LASTOUTPUT" />
-        <echo message="${LASTOUTPUT}" />
-      </then>
+        <exec command="${apachectl} restart" passthru="true" />
+        </then>
```
Comment (demiankatz) [inline build.xml]: Looks like indentation is off on this line.

File: build.xml [hunk 3]
```diff
@@ -439,18 +466,44 @@
...
+    <if>
+      <available file="${srcdir}/composer.phar" />
+      <then>
+        <delete file="${srcdir}/composer.phar" failonerror="false" />
+      </then>
+    </if>
+    <if>
+      <available file="${srcdir}/vufind-exception.log" />
+      <then>
+        <delete file="${srcdir}/vufind-exception.log" failonerror="false" />
```
Comment (demiankatz) [inline build.xml]: Again, probably safe to fail on error if we're checking for availability first.

**Review by demiankatz — COMMENTED** [review]
Thanks for this, @EreMaijala, it looks like a great improvement. I haven't had time to fully review it (very busy day today), but I left a couple of very small suggestions. I'll take a closer look as soon as time permits.

**Comment by EreMaijala** [comment]
@demiankatz Thanks, comments should be addressed now.

**Comment by demiankatz** [comment]
Looks good to me, and tests are all still functioning as expected. Merging now -- thanks for all the work on this!

**Pull request merged**